test/testcondition.cpp

```diff
@@ -1300,6 +1301,39 @@ class TestCondition : public TestFixture {
         ASSERT_EQUALS("[test.cpp:2] -> [test.cpp:3] -> [test.cpp:5]: (warning) Logical conjunction always evaluates to false: a > x && a < y.\n", errout.str());
     }
 
+    void incorrectLogicOperator13() {
+        // 8780
+        check("void f(const int &v) {\n"
+              "    const int x=v;\n"
+              "    if ((v == 1) && (x == 2)) {;}\n"
+              "    if ((x == 1) && (x == 2)) {;}\n"
+              "    if ((x == 1) && (v == 2)) {;}\n"
+              "    if ((v == 1) && (v == 2)) {;}\n"
+              "}\n");
+        ASSERT_EQUALS("[test.cpp:2] -> [test.cpp:3]: (warning) Logical conjunction always evaluates to false: v == 1 && x == 2.\n"
+                      "[test.cpp:4]: (warning) Logical conjunction always evaluates to false: x == 1 && x == 2.\n"
+                      "[test.cpp:2] -> [test.cpp:5]: (warning) Logical conjunction always evaluates to false: x == 1 && v == 2.\n"
+                      "[test.cpp:6]: (warning) Logical conjunction always evaluates to false: v == 1 && v == 2.\n"
+                      "[test.cpp:2] -> [test.cpp:3] -> [test.cpp:4]: (style) The if condition is the same as the previous if condition\n"
```

Comment:
I think it would be better to separate each diagnostic into separate checks or separate functions instead of adding this diagnostic.

Reply:
Good point, I agree completely. Let's not merge this until I have split it up.